src/asmcnc/core_UI/hoverable.py

```diff
@@ -0,0 +1,500 @@
+            elif key == 276:  # left
+                self.widget.pos[0] -= self.step_width
+            Logger.debug('pos: {}'.format(self.widget.pos))
+        except Exception as ex:
+            Logger.error("Failed to move widget!")
+
+    def get_parent_screen(self):
+        """Returns the screen object for the selected widget."""
+        p = self.widget
+        while not issubclass(type(p), Screen):
+            p = p.parent
+        return p
+
+    def dump_screen(self):
+        self.get_parent_screen().dump_screen()
+
+    def dump_screen_tsv(self):
+        j = self.get_parent_screen().dump_screen()
+
+    def print_help(self):
+        Logger.debug('Walking through the widget tree:')
+        Logger.debug('--------------------------------')
+        Logger.debug('l: (un)locks the selected widget.')
+        Logger.debug('e: (e)dit_mode on/off (enables moving widgets)')
+        Logger.debug('i: inspects the widget and shows its attributes.')
+        Logger.debug('c: switches to the first child.')
+        Logger.debug('s: cycles through siblings. You need to switch to a child first!')
+        Logger.debug('a: cycles through siblings backwards. You need to switch to a child first!')
+        Logger.debug('p: switches to the parent.')
+        Logger.debug('[del]: deletes the selected widget.')
+        Logger.debug('+/-: in-/decreases the step width for moving widgets with arrow keys')
+        Logger.debug('w: Open/Close popup to add new widgets to the currently selected one.')
+        Logger.debug('t: Tetris mode. snaps widgets to the closest one in the given direction (with padding).')
+        Logger.debug('When a widget is selected and edit_mode = True, you can use the arrow keys to move it.')
+        Logger.debug("Or just drag'n drop it with the mouse")
+
+    def get_widget_name_class(self, widget):
+        """
+        Tries to get the name of the given widget.
+        If the widget has no attribute 'name' or 'id' it uses 'NO_NAME' instead.
+
+        returns name (class name)
+        """
+        # check if it has a name
+        if hasattr(widget, 'name'):
+            name = widget.name
+        elif hasattr(widget, 'id') and widget.id:
+            name = widget.id
+        else:
+            name = 'NO_NAME'
+        return '{} ({})'.format(name, type(widget).__name__)
+
+    def switch_to_sibling(self):
+        """
+        Switches to the next sibling. If there are no more siblings left it returns to the first one.
+        This only works if switched to a child before to get the number of siblings.
+        """
+        if self.child_max > 1:
+            if self.child_index < self.child_max - 1:
+                self.child_index += 1
+            else:
+                self.child_index = 0
+            self.widget = self.widget.parent.children[self.child_index]
+            try:
+                # check if it is a Layout and has no name
+                name = self.widget.name
```

Comment:
if hasattr(widget, 'name'):

Reply:
removed try-except completely as I'm already calling the get_widget_name_class function. 'name' wasn't used anymore.